hubverse_annotator/utils.py

```diff
@@ -131,6 +131,65 @@ def get_reference_dates(forecast_table: pl.DataFrame) -> list[datetime.date]:
     return forecast_table.get_column("reference_date").unique().to_list()
 
 
+def get_initial_window_range(
+    data_to_plot: pl.DataFrame,
+    forecast_to_plot: pl.DataFrame,
+    observed_date_col: str = "date",
+    forecast_date_col: str = "target_end_date",
+    extra_weeks: int = 6,
+) -> list[datetime.datetime, datetime.datetime]:
+    """
+    Compute an initial x-axis window for plotting of
+    forecasts.
+
+    Parameters
+    ----------
+    data_to_plot : pl.DataFrame
+        Hubverse-formatted observed time-series, filtered
+        to the requested location, target, and models.
+    forecast_to_plot : pl.DataFrame
+        Hubverse-formatted forecast table, filtered to the
+        requested location, target, and models.
+    observed_date_col : str, optional
+        Name of the date column in `data_to_plot`
+        Defaults to "date".
+    forecast_date_col : str, optional
+        Name of the date column in `forecast_to_plot`.
+        Defaults to "target_end_date"`.
+    extra_weeks : int, optional
+        How many weeks before the first forecast to
+        include. Defaults to 6.
+
+    Returns
+    -------
+    list[datetime.datetime, datetime.datetime]
+        A 2-tuple `(start, end)` giving the initial
+        plotting window for forecast viewing.
+    """
+    data_to_plot = data_to_plot.with_columns(
+        pl.col(observed_date_col).cast(pl.Datetime)
+    )
+    forecast_to_plot = forecast_to_plot.with_columns(
+        pl.col(forecast_date_col).cast(pl.Datetime)
+    )
```

Comment:
Same here. These should be guaranteed datetimes, since we have the schema check.